File: backend/app/schemas/bank_account.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class BankAccountCreate(BaseModel):
    bank_name: str
    account_holder_name: str | None = None
    account_no: str | None = None
    account_type: str | None = "current"
    ifsc_code: str | None = None
    branch: str | None = None
    opening_balance: float | None = 0
    notes: str | None = None


class BankAccountUpdate(BaseModel):
    bank_name: str | None = None
    account_holder_name: str | None = None
    account_no: str | None = None
    account_type: str | None = None
    ifsc_code: str | None = None
    branch: str | None = None
    opening_balance: float | None = None
    current_balance: float | None = None
    notes: str | None = None
    is_active: bool | None = None


class BankAccountResponse(BaseModel):
    id: uuid.UUID
    bank_name: str
    account_holder_name: str | None
    account_no: str | None
    account_type: str | None
    ifsc_code: str | None
    branch: str | None
    opening_balance: float | None
    current_balance: float | None
    notes: str | None
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: backend-spring/src/main/java/com/nearkart/entity/BankAccount.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "bank_accounts")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class BankAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "bank_name", nullable = false)
    private String bankName;

    @Column(name = "account_holder_name")
    private String accountHolderName;

    @Column(name = "account_no")
    private String accountNo;

    @Column(name = "account_type")
    @Builder.Default
    private String accountType = "current";

    @Column(name = "ifsc_code")
    private String ifscCode;

    private String branch;

    @Column(name = "opening_balance", precision = 14, scale = 2)
    @Builder.Default
    private BigDecimal openingBalance = BigDecimal.ZERO;

    @Column(name = "current_balance", precision = 14, scale = 2)
    @Builder.Default
    private BigDecimal currentBalance = BigDecimal.ZERO;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() { OffsetDateTime now = OffsetDateTime.now(); createdAt = now; updatedAt = now; }

    @PreUpdate
    protected void onUpdate() { updatedAt = OffsetDateTime.now(); }
}
